core/dropfile.js

```diff
@@ -68,185 +82,141 @@ module.exports = class DropFile {
     }
 
     getHandler() {
-        return {
-            DOOR: this.getDoorSysBuffer,
-            DOOR32: this.getDoor32Buffer,
-            DORINFO: this.getDoorInfoDefBuffer,
-        }[this.fileType];
+        // TODO: Replace with a switch statement once we have binary handlers as well
+
+        // Read the directory containing the dropfile formats, and return undefined if we don't have the format
+        const fileName = this.fileName;
+        if (!fileName) {
+            Log.info({fileType: this.fileType}, 'Dropfile format not supported.');
+            return undefined;
+        }
+        const filePath = paths.join(this.dropFileFormatDirectory, fileName);
+        if(!fs.existsSync(filePath)) {
+            Log.info({filename: fileName}, 'Dropfile format not found or readable.');
+            return undefined;
+        }
+
+        // Return the handler to get the dropfile, because in the future we may have additional handlers
+        return this.getDropfile;
     }
 
     getContents() {
-        const handler = this.getHandler().bind(this);
-        return handler();
+        const handlerRef = this.getHandler();
+        if(!handlerRef) {
+            return undefined;
+        }
+        const handler = handlerRef.bind(this);
+        const contents = handler();
+        return contents;
     }
 
-    getDoorInfoFileName() {
-        let x;
-        const node = this.client.node;
-        if (10 === node) {
-            x = 0;
-        } else if (node < 10) {
-            x = node;
-        } else {
-            x = String.fromCharCode('a'.charCodeAt(0) + (node - 11));
+    getDropfile() {
+        // Get the filename to read
+        const fileName = paths.join(this.dropFileFormatDirectory, this.fileName);
+
+        let text = fs.readFileSync(fileName);
+
+        // Format the data with string_format and predefined_mci
+        let formatObj = getPredefinedMCIFormatObject(this.client, text);
+
+        const additionalFormatObj = {
+            'getSysopFirstName': this.getSysopFirstName(),
+            'getSysopLastName': this.getSysopLastName(),
+            'getUserFirstName': this.getUserFirstName(),
+            'getUserLastName': this.getUserLastName(),
+            'getUserTotalDownloadK': this.getUserTotalDownloadK(),
```

Comment:
I *think* `{userTotalDownloadBytes!toKilobytes}` (by resolving some TODO's in `string_format.js` would work here and be more flexible?

Reply:
So this one and the date format I had thought about, but there is a bit of a problem with when these methods are called... it looks like the methods run first and return a string, then the formatting runs on them. So for these it would actually have to parse as a string to format, which is a bit ugly. Maybe a better way would be to have these return an object with a toString instead, then we can format them as we like, with the default behaving as it does now? Just a thought.

Reply:
I'll take a look at this maybe tomorrow night. I think in `string_util.js` it can be expanded to allow some Date/Time raw formats of some sort and via transformers:

```javascript
            try {
                value = getValue(obj, objPath);
                if (transformer) {
// 'value' can be a obj here IIRC
                    value = transformValue(transformer, value);
                }

                tokens = tokenizeFormatSpec(formatSpec || '');
// could allow non-transformers to handle D/T here too... maybe
                if (_.isNumber(value)) {
                    out += formatNumber(value, tokens);
                } else {
                    out += formatString(value, tokens);
                }
            } catch (e) {
                if (e instanceof KeyError) {
                    out += match[0]; //  preserve full thing
                } else if (e instanceof ValueError) {
                    out += value.toString();
                }
            }
```